node/pkg/watchers/hedera/watcher.go

```diff
@@ -0,0 +1,612 @@
+	go func() {
+		ticker := time.NewTicker(5 * time.Second)
+		client := &http.Client{
+			Timeout: time.Second * 5,
+		}
+
+		// Do not add a leading slash
+		latestBlockURL := "api/v1/blocks"
+
+		for {
+			<-ticker.C
+			// Query and report height and set currentSlotHeight
+			logger.Info("Checking the following", zap.String("urlLCD", watcher.urlLCD), zap.String("latestBlockURL", latestBlockURL))
+			resp, err := client.Get(fmt.Sprintf("%s/%s", watcher.urlLCD, latestBlockURL))
+			if err != nil {
+				logger.Error("query latest block response error", zap.String("network", networkName), zap.Error(err))
+				continue
+			}
+			blocksBody, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logger.Error("query latest block response read error", zap.String("network", networkName), zap.Error(err))
+				errC <- err
+				resp.Body.Close()
+				// When this thread goes away due to error, shouldn't the other threads go away, too?
+				break
+			}
+			resp.Body.Close()
+
+			blockJSON := string(blocksBody)
+			latestBlock := gjson.Get(blockJSON, "blocks.0.number")
+			logger.Info("current height", zap.String("network", networkName), zap.Int64("block", latestBlock.Int()))
+			p2p.DefaultRegistry.SetNetworkStats(watcher.chainID, &gossipv1.Heartbeat_Network{
+				Height:          latestBlock.Int(),
+				ContractAddress: watcher.contract,
+			})
+		}
+	}()
+
+	// This function is for reobservations
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-watcher.obsvReqC:
+				if vaa.ChainID(r.ChainId) != watcher.chainID {
+					panic("invalid chain ID")
+				}
+
+				tx := hex.EncodeToString(r.TxHash)
+
+				logger.Info("received observation request", zap.String("network", networkName), zap.String("tx_hash", tx))
+
+				client := &http.Client{
+					Timeout: time.Second * 5,
+				}
+
+				// Query for tx by hash
+				hashString := "api/v1/contracts/results/" + tx
+				resp, err := client.Get(fmt.Sprintf("%s/%s", watcher.urlLCD, hashString))
+				if err != nil {
+					logger.Error("query tx response error", zap.String("network", networkName), zap.Error(err))
+					continue
+				}
+				txBody, err := io.ReadAll(resp.Body)
+				if err != nil {
+					logger.Error("query tx response read error", zap.String("network", networkName), zap.Error(err))
+					resp.Body.Close()
+					continue
+				}
+				resp.Body.Close()
+
+				txJSON := string(txBody)
+				logger.Info("RO: txJSON", zap.String("txJSON", txJSON))
+
+				events, err := TxMessageToEvent(logger, txJSON)
+
+				for _, ev := range events {
+					watcher.msgChan <- ev
+					// messagesConfirmed.WithLabelValues(networkName).Inc()
+				}
+			}
+		}
+	}()
+
+	// This function is for normal watcher operations
+	go func() {
+		defer close(errC)
+		t := time.NewTicker(5 * time.Second)
+		client := &http.Client{
+			Timeout: time.Second * 5,
+		}
+		var beginningTS = "1234567890.000000400" // TODO:  Should this be Time.now()?
```

Comment:
Yes, I think it should be the time at startup.

Reply:
Fixed